src/qml/main.qml

```diff
@@ -80,6 +114,36 @@ CutieWindow {
                 elide: Text.ElideRight
                 horizontalAlignment: Text.AlignHCenter
             }
+            
+            CutieStore {
+              id: favoriteStore
+              appName: "cutie-launcher"
+              storeName: "favoriteItems"
+            }
+
+            function saveFavoriteItem(name, iconPath, execCommand) {
+               let data = favoriteStore.data;
+               data["favoriteApp-" + name] = { "icon": iconPath, "command": execCommand };
```

Comment:
There is really no need to prefix it like this. Also, it would make sense the save the full js object (model).

Reply:
Got it .. I saw that the add data function only uses these 3 variables to create the deligate item, that's why I did the same.. should I be saving the whole object?

Reply:
Actually, the best way to do this imo would be to save the filename of the desktop file.

So, for example, when adding Firefox from `/usr/share/applications/firefox-esr.desktop`, the code should add `"firefox-esr"` to a JavaScript array.

With this, I would rename the store to `settings` and use `data["favoriteApps"]` as the array.

Reply:
Is the intension behind this is to have an extra page in the settings app to manage apps?

Also if we only save the file name then won't we need a lot more code to read the desktop file to get the icon and app name??

(Sorry if I miss understand your idea)

Reply:
Well, it is just redundant to store all information twice.

With this, it would be easier to remove uninstalled applications from favorites automatically.

In this case, it would be sane to move the desktop file parsing logic to a library.

Reply:
So the .desktop file is phrased and sent to libcutiestore then both all apps and favourite apps are loaded from this store?

Reply:
No, the desktop entries would be parsed on the fly. Of favorites, only the file name of the entry is saved. Then, cutie-home would parse the desktop file of this name.

Reply:
Oke, I think I understand  now. will come back to this once I am done with cutie-home

Reply:
hi @erikinkinen  how hard would it be to move  desktop file parsing logic to a library.